src/client/js/section-observer.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const classNameToObserve = 'section-transition'
+const classNameEntered = `${classNameToObserve}-entered`
+const cueIntervalDuration = 150
+const sectionThreshold = 0.1
+
+let observers
+let cueInterval
+const entryCue = []
+
+function handleShowSection () {
+  if (!entryCue?.length && cueInterval) {
+    clearInterval(cueInterval)
+    cueInterval = null
+    return
+  }
+
+  const nextEntry = entryCue.shift()
+  nextEntry.target.classList.add(classNameEntered)
+}
+
+function setCueInterval () {
+  cueInterval = setInterval(handleShowSection, cueIntervalDuration)
+}
+
+function handleScroll (entries) {
+  entries.forEach(entry => {
+    const sectionElement = entry.target
+    const hasEntered = sectionElement.classList.contains(classNameEntered)
+
+    if (hasEntered) {
+      return
+    }
+
+    const isInViewport = entry.isIntersecting
+    if (isInViewport) {
+      entryCue.push(entry)
+    }
+
+    if (!cueInterval) {
+      setCueInterval()
+    }
+  })
+}
+
+function init (sections) {
+  const observer = new IntersectionObserver(handleScroll, {
+    threshold: sectionThreshold
+  })
+
+  observers = [...sections].map(section => {
+    observer.observe(section)
+    return observer
+  })
+
+  setCueInterval()
+}
+
+if (!observers) {
+  const mediaQuery = window.matchMedia('(prefers-reduced-motion: no-preference)')
+  const allowMotion = mediaQuery && mediaQuery.matches
+
+  const sections = document.getElementsByClassName(classNameToObserve)
+  mediaQuery.addEventListener('change', () => {
+    [...sections].forEach(section => section.classList.add(classNameEntered))
+  })
```

Comment:
Handle the following edge case: A user has the Monitor landing page open and is setting their preferences.

Reply:
Not sure I understand this edge case 😊

Reply:
The edge case is:
- a user loaded the Monitor site with their motion preferences set to “reduce”
- has the Monitor site open and sets the preferences back to “no-preference”
- in that case the page would be blank until the next reload

Instead of setting the class names I’m setting the CSS variables now.

https://user-images.githubusercontent.com/13835474/232039492-b1a91af7-62bf-4cb7-a754-3f08ec0d0456.mov